pyccel/ast/variable.py

```diff
@@ -357,6 +351,14 @@ def is_ndarray(self):
     def __str__(self):
         return str(self.name)
 
+    def __eq__(self, other):
+        if type(self) is type(other):
+            return self._name == other.name
+        return False
+
+    def __hash__(self):
+        return hash((type(self).__name__, self._name))
```

Comment:
Just curious, why do you use `type(self).__name__` instead of just `type(self)`?

Reply:
I just prefered to hash string  because I don't know how 'class type' hash behave

Reply:
I see, thanks!